base/REPLCompletions.jl

```diff
@@ -161,7 +161,13 @@ function complete_path(path::AbstractString, pos; use_envpath=false)
                 continue
             end
 
-            local filesinpath = readdir(pathdir)
+            local filesinpath
+            try
+                filesinpath = readdir(pathdir)
+            catch
+                # Bash allows dirs in PATH that can't be read, so we should as well.
```

Comment:
this should be more specific about only catching and continuing in the case of the specific expected exception type